File: 0206-reverse-linked-list/0206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        #1. if head is none return none 
        if head == None:
            return None
        # Reverse a linked list using cnpc
        currentPointer = head
        previousPointer = None
        
        while currentPointer:
            temp = currentPointer.next 
            currentPointer.next = previousPointer
            previousPointer = currentPointer
            currentPointer = temp
        return previousPointer

File: 0206-reverse-linked-list/0206-reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        //Check if the list node is empty 
        if(head == null){
            return null;
        }
        
        ListNode previous = null;
        ListNode current = head;
        //cnpc
        while (current!=null){
            ListNode temp = current.next;
            current.next = previous;
            previous = current;
            current = temp;
        }
        
        return previous;
        
    }
}